assets/global.js

```diff
@@ -934,7 +935,13 @@ class VariantSelects extends HTMLElement {
 
         if (inventoryDestination) inventoryDestination.classList.toggle('visibility-hidden', inventorySource.innerText === '');
 
-        this.toggleAddButton(!this.currentVariant.available, window.variantStrings.soldOut);
+        const addButtonUpdated = html.getElementById(`ProductSubmitButton-${sectionId}`);
+        this.toggleAddButton(addButtonUpdated ? addButtonUpdated.hasAttribute('disabled') : true, window.variantStrings.soldOut);
+
+        publish(PUB_SUB_EVENTS.variantChange, {data: {
+          sectionId,
+          html
+        }});
```

Comment:
I think if you add something called `variantChange` to a global pub/sub, a lot of folks in the ecosystem will subscribe to it expecting useful information that will inform their subscription apps, bundle popups, custom carts, sizing widgets, and so on. They are going to be super disappointed when there's no `data.variant` inside.

Reply:
Good point. Updated - https://github.com/Shopify/dawn/pull/2251/commits/c668a7beac728fd6a5438546c678610a670760a7